### ISSUE TITLE
Syncing project.yaml changes causes writes to project.yaml while in read only mode 

### ISSUE THREAD

**Issue by mrksmts** [spicywebau/craft-neo#217]
#### Description
When applying the project.yaml changes Neo tries to update the project.yaml, even though it is in read-only mode.

#### Steps to reproduce
1. Syncing project.yaml changes

#### Other information
- Neo version: 2.3.4
- Craft version: Craft Pro 3.1.28
- Is the affected Craft installation multi-site: No

From the logs:
```
2019-05-23 11:06:33 [-][-][-][error][yii\base\NotSupportedException] yii\base\NotSupportedException: Changes to the project config are not possible while in read-only mode. in /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php:386
Stack trace:
#0 /home/****/vendor/spicyweb/craft-neo/src/services/BlockTypes.php(242): craft\services\ProjectConfig->set('neoBlockTypes.0...', Array)
#1 /home/****/vendor/spicyweb/craft-neo/src/services/Fields.php(111): benf\neo\services\BlockTypes->save(Object(benf\neo\models\BlockType), false)
#2 /home/****/vendor/spicyweb/craft-neo/src/Field.php(585): benf\neo\services\Fields->save(Object(benf\neo\Field))
#3 /home/****/vendor/craftcms/cms/src/services/Fields.php(1554): benf\neo\Field->afterSave(false)
#4 /home/****/vendor/craftcms/cms/src/services/Fields.php(819): craft\services\Fields->applyFieldSave('7bac5357-5482-4...', Array, 'global')
#5 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(884): craft\services\Fields->handleChangedField(Object(craft\events\ConfigEvent))
#6 [internal function]: craft\services\ProjectConfig->craft\services\{closure}(Object(craft\events\ConfigEvent))
#7 /home/****/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Object(Closure), Object(craft\events\ConfigEvent))
#8 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(585): yii\base\Component->trigger('updateItem', Object(craft\events\ConfigEvent))
#9 /home/****/vendor/craftcms/cms/src/helpers/ProjectConfig.php(64): craft\services\ProjectConfig->processConfigChanges('fields.7bac5357...')
#10 /home/****/vendor/craftcms/cms/src/services/Users.php(1044): craft\helpers\ProjectConfig::ensureAllFieldsProcessed()
#11 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(884): craft\services\Users->handleChangedUserFieldLayout(Object(craft\events\ConfigEvent))
#12 [internal function]: craft\services\ProjectConfig->craft\services\{closure}(Object(craft\events\ConfigEvent))
#13 /home/****/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Object(Closure), Object(craft\events\ConfigEvent))
#14 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(585): yii\base\Component->trigger('updateItem', Object(craft\events\ConfigEvent))
#15 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(878): craft\services\ProjectConfig->processConfigChanges('users.fieldLayo...')
#16 [internal function]: craft\services\ProjectConfig->craft\services\{closure}(Object(craft\events\ConfigEvent))
#17 /home/****/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Object(Closure), Object(craft\events\ConfigEvent))
#18 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(572): yii\base\Component->trigger('removeItem', Object(craft\events\ConfigEvent))
#19 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(943): craft\services\ProjectConfig->processConfigChanges('users.fieldLayo...')
#20 /home/****/vendor/craftcms/cms/src/services/ProjectConfig.php(467): craft\services\ProjectConfig->_applyChanges(Array)
#21 /home/****/vendor/craftcms/cms/src/controllers/ConfigSyncController.php(57): craft\services\ProjectConfig->applyYamlChanges()
#22 [internal function]: craft\controllers\ConfigSyncController->actionApplyYamlChanges()
#23 /home/****/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#24 /home/****/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#25 /home/****/vendor/craftcms/cms/src/web/Controller.php(109): yii\base\Controller->runAction('apply-yaml-chan...', Array)
#26 /home/****/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('apply-yaml-chan...', Array)
#27 /home/****/vendor/craftcms/cms/src/web/Application.php(297): yii\base\Module->runAction('config-sync/app...', Array)
#28 /home/****/vendor/craftcms/cms/src/web/Application.php(726): craft\web\Application->runAction('config-sync/app...')
#29 /home/****/vendor/craftcms/cms/src/web/Application.php(248): craft\web\Application->_processConfigSyncLogic(Object(craft\web\Request))
#30 /home/****/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#31 /home/****/web/index.php(22): yii\base\Application->run()
#32 {main}
2019-05-23 11:06:32 [-][-][-][info][application] $_GET = [
    'p' => 'admin/actions/config-sync/apply-yaml-changes'
]
```

**Comment by texasbaptists** [comment]
I believe this is happening with my project too. When pushing a change to a staging server, Forge send this error:
```
Yii Migration Tool (based on Yii v2.0.21)

Migrated up successfully.
Applying changes from project.yaml ... error: Changes to the project config are not possible while in read-only mode.
```

**Comment by mihow** [comment]
We are running into this as well

**Comment by pvldigital** [comment]
Hi all,

Fix has been implemented in release 2.5.10.

Report back if there's any issues and i'll reopen the ticket.
